docs/configure/agent.md

```diff
@@ -0,0 +1,311 @@
+
+### `viam-server`
+
+<!-- prettier-ignore -->
+| Option | Type | Required? | Description |
+| ------ | ---- | --------- | ----------- |
+| `attributes` | object | Optional | <ul><li>`fast_start`: If set to `true`, `viam-agent` will not wait for a network connection nor check for updates before starting `viam-server`. See [Fast start mode](#fast-start-mode).</li></ul> |
+| `release_channel` | string | Optional | `viam-agent` is semantically versioned and is tested before release. Releases happen infrequently. When set to `"stable"`, `viam-agent` will automatically upgrade when a new version is released. Options: `"stable"` (default), `"latest"`. |
+| `pin_version` | string | Optional | "Lock" the subsystem to a specific version (as provided by the release channel). If set, no automatic upgrades will be performed until the setting is updated to a new version (or removed to revert to the release channel). If both `pin_url` and `pin_version` is set, `pin_url` will be used. Default: `""`. |
+| `pin_url` | string | Optional | Ignore normal version selection and directly download from the specified URL. If set, no automatic upgrades will be performed until the setting is updated to a new URL (or removed to revert to the release channel). Typically this is only used for testing/troubleshooting. If both `pin_url` and `pin_version` is set, `pin_url` will be used. Default: `""`. |
+| `disable_subsystem` | boolean | Optional | When set to `true` it disables the `viam-server` subsystem. |
+
+#### Fast start mode
+
+You can use fast start mode to bypass `viam-agent` waiting for a network connection to be established and checking for updates during initial startup.
+This will result in `viam-server` executing as quickly as possible.
+
+This is useful if you have a device that often starts when offline or on a slow connection, and if having the latest version immediately after start isn't required.
+
+{{< alert title="Note" color="note" >}}
+Period update checks will continue to run afterwards.
+The fast start mode only affects the initial startup sequencing.
+{{< /alert >}}
+
+You can also start `viam-agent` in fast start mode by setting `VIAM_AGENT_FAST_START=1` in your environment.
+
+### `agent-provisioning`
+
+<!-- prettier-ignore -->
+| Option | Type | Required? | Description |
+| ------ | ---- | --------- | ----------- |
+| `attributes` | object | Optional | <ul><li>`hotspot_password`: Set a password for the WiFi hotspot a machine creates during provisioning.</li><li>`networks`: Networks a machine can automatically connect to when roaming mode is enabled in the [`agent-provision`](/fleet/provision/#configuration) and either no primary network was configured by the end user in the provisioning app or the primary network cannot be connected to. See [#networks].</li><li>`roaming_mode`: If enabled, lets the machine connect to additional configured netowrks. See [Networks](#networks).</li></ul> |
```

Comment:
@Otterverse can you check these? Is hotspot_password meant to be configurable here? Maybe I put that here wrong.

Reply:
Under the hood, everything in provisioning's "attributes" is/can be identical to that in "/etc/viam-provisioning.json" The on-machine file is meant to be customized by the manufacturer, to provide "out of the box" settings. The attributes (in the cloud config) can let the user override those if they wish, but (as it's in the cloud) only after the machine gets online. Most of the initial stuff (like manufactuer, model, fragment_id, etc) the user generally shouldn't change, but the hotspot_password is one they can/should (for security), along with roaming_mode and any additional networks they want to configure. Anything here overrides what is set by the manufacturer in "/etc/viam-provisioning.json"

Reply:
Ok, I'll add that it overrides those settings and add the rest of the settings I guess.

Reply:
done.

Reply:
Sorry, wasn't saying you need to add that specifically. Was clarifying for YOUR understanding primarily. The reason for only listing SOME of the available options here is that it's only recommended to change these (for the end user.)

Reply:
I understand - I think if I didn't understand it other might not either, so I think adding it might be good